sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py

```diff
@@ -454,15 +454,16 @@ def create_snapshot( # type: ignore
 
     @distributed_trace
     def delete_share(
-            self, delete_snapshots=False, # type: Optional[bool]
-            **kwargs
-        ):
-        # type: (...) -> None
+        self, delete_snapshots: Optional[Union[Literal['include'], Literal['include-leased']]],
```

Comment:
This type can be simplified since `Literal` can accept multiple values. `Optional[Literal['include', 'include-leased']]`